### ISSUE TITLE
How to compare two image

### ISSUE THREAD

**Issue by swapnilyadavmpsedc** [tensorflow/tensorflow#62952]
How to compare two image using  tensorflow

**Comment by tilakrayal** [comment]
@swapnilyadavmpsedc,
There are multiple options for the similar taks. One of them is extracting features from images using pre-trained models like **VGG16** or **Inceptionv3** and then comparing those features with metrics like cosine similarity or Euclidean distance.

https://www.tensorflow.org/api_docs/python/tf/keras/applications/vgg16/VGG16
https://www.tensorflow.org/api_docs/python/tf/keras/applications/inception_v3/InceptionV3

Also you can try training a **Siamese network** with image pairs and labels (similar/dissimilar). This approach learns complex similarity relationships. A Siamese Network is a type of network architecture that contains two or more identical subnetworks used to generate feature vectors for each input and compare them.

https://keras.io/examples/vision/siamese_network/

Thank you!
